### PR TITLE
test: reproduce 10582

### PR DESCRIPTION
**Pull request by mt-caret**
Reproduction of https://github.com/ocaml/dune/issues/10582.

### CODE REVIEW

**Review by anmonteiro — APPROVED** [review]

File: test/blackbox-tests/test-cases/mdx-stanza/shared-libraries.t/dune [hunk 1]
```diff
@@ -0,0 +1,16 @@
+(library
+ (name public_lib)
+ (foreign_archives test))
+
+(rule
+ (deps test.c)
+ (targets libtest.a dlltest.so)
+ (action
+  (progn
+   (run gcc -c -fPIC test.c -o test.o)
```
Comment (rgrinberg) [inline test/blackbox-tests/test-cases/mdx-stanza/shared-libraries.t/dune:10]: The test should depend on the `gcc` binary and `ar` binaries.

**Review by rgrinberg — APPROVED** [review]
@anmonteiro feel free to merge if you understand the issue. I added some comments about missing dependencies though

File: test/blackbox-tests/test-cases/mdx-stanza/shared-libraries.t/dune (hunk 1 above)
Comment (anmonteiro) [reply test/blackbox-tests/test-cases/mdx-stanza/shared-libraries.t/dune:10]: I fixed it in #10586

**Comment by anmonteiro** [comment]
Continued in #10586 because this test fails on macOS.

**Pull request closed without merging**